### PR TITLE
TimerTrigger DST fix (#2546)

### PR DESCRIPTION
**Pull request by mathewc**
Fix for https://github.com/Azure/azure-functions-host/issues/2546 

Fix actually identified by @npiasecki in the issue thread (Thanks!) I've taked the suggested fix and added regression tests for this.

### CODE REVIEW

File: test/WebJobs.Extensions.Tests/Extensions/Timers/Listener/TimerListenerTests.cs [hunk 1]
```diff
@@ -355,6 +356,61 @@ public void Timer_VerifyMaxInterval()
...
+        /// Situation where the DST transition happens in the middle of the schedule, with the
+        /// next occurrence AFTER the DST transition.
+        /// </summary>
+        [Fact]
+        public void GetNextInterval_NextAfterDST_ReturnsExpectedValue()
+        {
+            // Running on the Friday before the DST switch at 2 AM on 3/11 (Pacific Standard Time)
+            // Note: this test uses Local time, so if you're running in a timezone where
+            // DST doesn't transition the test might not be valid.
+            var now = new DateTime(2018, 3, 9, 18, 0, 0, DateTimeKind.Local);
```
Comment (brettsam) [inline test/WebJobs.Extensions.Tests/Extensions/Timers/Listener/TimerListenerTests.cs]: Will this pass in AppVeyor? I believe it uses UTC.

**Review by brettsam — COMMENTED** [review]
Does the ScaleController need this logic?

**Comment by mathewc** [comment]
@cgillum for the ScaleController question

File: test/WebJobs.Extensions.Tests/Extensions/Timers/Listener/TimerListenerTests.cs (hunk 1 above)
Comment (mathewc) [reply test/WebJobs.Extensions.Tests/Extensions/Timers/Listener/TimerListenerTests.cs]: Yes they will pass there. Looks like my PR build passed?

**Comment by ToastHawaii** [comment]
Hello mathewc
Is there a version 2.x with this fix? The last Nuget is 2.2 and the commit "b427130 " with this Tag is from 12 Mar. Does it contain this fix?
Thanks

**Pull request merged**